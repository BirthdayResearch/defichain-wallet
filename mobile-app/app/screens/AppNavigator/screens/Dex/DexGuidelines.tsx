import { MaterialIcons } from '@expo/vector-icons'
import * as React from 'react'
import { View } from '@components/index'
import { Button } from '@components/Button'
import { ThemedIcon, ThemedScrollView, ThemedText } from '@components/themed'
import { tailwind } from '@tailwind'
import { translate } from '@translations'

interface Props {
  onClose: () => void
}

interface GuidelineItem {
  icon: React.ComponentProps<typeof MaterialIcons>['name']
  title: string
  subtitle: string
}

// TODO (Harsh): handle language change bug, when user change the language, sometime it didnt get update the satic page
export function DexGuidelines ({ onClose }: Props): JSX.Element {
  const guidelines: GuidelineItem[] = [
    {
      title: 'Add liquidity',
      subtitle: 'Earn high yields by supplying token pairs to the liquidity pool.',
      icon: 'add'
    },
    {
      title: 'Swap tokens',
      subtitle: 'Conveniently swap participating tokens within the liquidity pool.',
      icon: 'swap-horiz'
    },
    {
      title: 'Withdraw at any time',
      subtitle: 'You have full control over your tokens unlike any other.',
      icon: 'account-balance-wallet'
    }
  ]

  return (
    <ThemedScrollView
      dark={tailwind('bg-gray-900')}
      light={tailwind('bg-white')}
      style={tailwind('flex-1 p-4 pt-6')}
      testID='dex_guidelines_screen'
    >
      <ThemedText
        style={tailwind('text-lg font-semibold')}
      >
        {translate('screens/DexGuidelines', 'Decentralized Exchange')}
      </ThemedText>

      <ThemedText
        dark={tailwind('text-gray-400')}
        light={tailwind('text-gray-500')}
        style={tailwind('mt-1 text-sm font-medium mb-4')}
      >
        {translate('screens/DexGuidelines', 'Participate in supplying liquidity to power the DEX (Decentralized Exchange). Use your tokens to earn high returns (of up to 100%).')}
      </ThemedText>

      {
        guidelines.map((g, i) => (
          <View
            key={i}
            style={tailwind('flex-row items-center my-4')}
          >
            <ThemedIcon
              iconType='MaterialIcons'
              name={g.icon}
              size={32}
            />

            <View style={tailwind('flex-col flex-auto ml-6')}>
              <ThemedText style={tailwind('font-medium')}>
                {translate('screens/DexGuidelines', g.title)}
              </ThemedText>

              <ThemedText
                dark={tailwind('text-gray-400')}
                light={tailwind('text-gray-500')}
                numberOfLines={4}
                style={tailwind('text-sm')}
              >
                {translate('screens/DexGuidelines', g.subtitle)}
              </ThemedText>
            </View>
          </View>
        ))
      }

      <Button
        label={translate('screens/DexGuidelines', 'CONTINUE')}
        margin='mx-0 mt-10 mb-8'
        onPress={onClose}
        testID='close_dex_guidelines'
        title={translate('screens/DexGuidelines', 'CONTINUE')}
      />
    </ThemedScrollView>
  )
}
