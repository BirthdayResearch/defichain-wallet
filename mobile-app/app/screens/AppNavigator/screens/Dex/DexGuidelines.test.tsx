import { render, waitFor } from '@testing-library/react-native'
import { DexGuidelines } from './DexGuidelines'

jest.mock('@shared-contexts/ThemeProvider')

describe('Dex guide', () => {
  it('should match snapshot', async () => {
    const onClose = jest.fn
    const rendered = render(
      <DexGuidelines onClose={onClose} />
    )
    await waitFor(() => rendered.toJSON() !== null)
    expect(rendered.toJSON()).toMatchSnapshot()
  })
})
