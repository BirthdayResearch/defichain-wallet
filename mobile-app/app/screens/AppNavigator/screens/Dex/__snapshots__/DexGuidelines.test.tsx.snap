// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dex guide should match snapshot 1`] = `
<RCTScrollView
  style={
    Array [
      Object {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 24,
      },
      Object {
        "backgroundColor": "rgba(255, 255, 255, 1)",
      },
    ]
  }
  testID="dex_guidelines_screen"
>
  <View>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "RegularFont",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
          Array [
            Object {
              "fontFamily": "SemiBoldFont",
              "fontSize": 18,
              "fontWeight": "600",
              "lineHeight": 28,
            },
            Object {
              "color": "rgba(0, 0, 0, 1)",
            },
          ],
        ]
      }
    >
      Decentralized Exchange
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "RegularFont",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
          Array [
            Object {
              "fontFamily": "MediumFont",
              "fontSize": 14,
              "fontWeight": "500",
              "lineHeight": 20,
              "marginBottom": 16,
              "marginTop": 4,
            },
            Object {
              "color": "rgba(115, 115, 115, 1)",
            },
          ],
        ]
      }
    >
      Participate in supplying liquidity to power the DEX (Decentralized Exchange). Use your tokens to earn high returns (of up to 100%).
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 16,
          "marginTop": 16,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 32,
            },
            Array [
              undefined,
              Object {
                "color": "rgba(0, 0, 0, 1)",
              },
            ],
            Object {
              "fontFamily": "material",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <View
        style={
          Object {
            "flexBasis": "auto",
            "flexDirection": "column",
            "flexGrow": 1,
            "flexShrink": 1,
            "marginLeft": 24,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "RegularFont",
                "fontSize": 16,
                "fontWeight": "400",
                "lineHeight": 24,
              },
              Array [
                Object {
                  "fontFamily": "MediumFont",
                  "fontWeight": "500",
                },
                Object {
                  "color": "rgba(0, 0, 0, 1)",
                },
              ],
            ]
          }
        >
          Add liquidity
        </Text>
        <Text
          numberOfLines={4}
          style={
            Array [
              Object {
                "fontFamily": "RegularFont",
                "fontSize": 16,
                "fontWeight": "400",
                "lineHeight": 24,
              },
              Array [
                Object {
                  "fontSize": 14,
                  "lineHeight": 20,
                },
                Object {
                  "color": "rgba(115, 115, 115, 1)",
                },
              ],
            ]
          }
        >
          Earn high yields by supplying token pairs to the liquidity pool.
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 16,
          "marginTop": 16,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 32,
            },
            Array [
              undefined,
              Object {
                "color": "rgba(0, 0, 0, 1)",
              },
            ],
            Object {
              "fontFamily": "material",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <View
        style={
          Object {
            "flexBasis": "auto",
            "flexDirection": "column",
            "flexGrow": 1,
            "flexShrink": 1,
            "marginLeft": 24,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "RegularFont",
                "fontSize": 16,
                "fontWeight": "400",
                "lineHeight": 24,
              },
              Array [
                Object {
                  "fontFamily": "MediumFont",
                  "fontWeight": "500",
                },
                Object {
                  "color": "rgba(0, 0, 0, 1)",
                },
              ],
            ]
          }
        >
          Swap tokens
        </Text>
        <Text
          numberOfLines={4}
          style={
            Array [
              Object {
                "fontFamily": "RegularFont",
                "fontSize": 16,
                "fontWeight": "400",
                "lineHeight": 24,
              },
              Array [
                Object {
                  "fontSize": 14,
                  "lineHeight": 20,
                },
                Object {
                  "color": "rgba(115, 115, 115, 1)",
                },
              ],
            ]
          }
        >
          Conveniently swap participating tokens within the liquidity pool.
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 16,
          "marginTop": 16,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 32,
            },
            Array [
              undefined,
              Object {
                "color": "rgba(0, 0, 0, 1)",
              },
            ],
            Object {
              "fontFamily": "material",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <View
        style={
          Object {
            "flexBasis": "auto",
            "flexDirection": "column",
            "flexGrow": 1,
            "flexShrink": 1,
            "marginLeft": 24,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "RegularFont",
                "fontSize": 16,
                "fontWeight": "400",
                "lineHeight": 24,
              },
              Array [
                Object {
                  "fontFamily": "MediumFont",
                  "fontWeight": "500",
                },
                Object {
                  "color": "rgba(0, 0, 0, 1)",
                },
              ],
            ]
          }
        >
          Withdraw at any time
        </Text>
        <Text
          numberOfLines={4}
          style={
            Array [
              Object {
                "fontFamily": "RegularFont",
                "fontSize": 16,
                "fontWeight": "400",
                "lineHeight": 24,
              },
              Array [
                Object {
                  "fontSize": 14,
                  "lineHeight": 20,
                },
                Object {
                  "color": "rgba(115, 115, 115, 1)",
                },
              ],
            ]
          }
        >
          You have full control over your tokens unlike any other.
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(255, 229, 247, 1)",
          "borderBottomLeftRadius": 4,
          "borderBottomRightRadius": 4,
          "borderTopLeftRadius": 4,
          "borderTopRightRadius": 4,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 32,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 40,
          "opacity": 1,
          "paddingBottom": 12,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 12,
        }
      }
      testID="close_dex_guidelines"
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RegularFont",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            Object {
              "color": "rgba(255, 0, 175, 1)",
              "fontFamily": "BoldFont",
              "fontWeight": "700",
              "textAlign": "center",
            },
          ]
        }
      >
        CONTINUE
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
